src/application/service/GenerateAuthToken.ts

```diff
@@ -0,0 +1,22 @@
+import { User } from '../../domain/model/user/User';
+import jwt from 'jsonwebtoken';
+
+export class GenerateAuthToken {
+  private jwtTimeToLiveInMs: number;
+
+  constructor(private jwtSecret: string, jwtTimeToLiveInDays: number) {
+    this.jwtTimeToLiveInMs = jwtTimeToLiveInDays * 24 * 60 * 60_000;
+  }
+
+  public async execute(user: User) {
+    return jwt.sign(
+      {
+        userId: user.id(),
+        userEmail: user.email(),
```

Comment:
Uhmm... I wouldn't leak the email in the token (as we are not using JWE) if we don't have any reason to use it